sdk/ai/ai-projects/samples-dev/agents/agentFileSearch.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * This sample demonstrates how to use the File Search Tool with streaming responses.
+ * It combines file search capabilities with response streaming to provide real-time
+ * search results from uploaded documents.
+ *
+ * @summary This sample demonstrates how to create an agent with file search capabilities,
+ * upload documents to a vector store, and stream responses that include file search results.
+ *
+ * @azsdk-weight 100
+ */
+
+import { DefaultAzureCredential } from "@azure/identity";
+import { AIProjectClient } from "@azure/ai-projects";
+import * as fs from "fs";
+import * as path from "path";
+import { fileURLToPath } from "url";
+import "dotenv/config";
+
+const projectEndpoint = process.env["AZURE_AI_PROJECT_ENDPOINT"] || "<project endpoint>";
+const deploymentName = process.env["MODEL_DEPLOYMENT_NAME"] || "<model deployment name>";
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const assetFilePath = path.resolve(__dirname, "assets", "product_info.md");
+
+export async function main(): Promise<void> {
+  // Create AI Project client
+  const project = new AIProjectClient(projectEndpoint, new DefaultAzureCredential());
+  const openAIClient = await project.getOpenAIClient();
+
+  console.log("Setting up file search with streaming responses...");
+
+  // Create vector store for file search
+  const vectorStore = await openAIClient.vectorStores.create({
+    name: "ProductInfoStreamStore",
+  });
+  console.log(`Vector store created (id: ${vectorStore.id})`);
+
+  // Upload file to vector store
+  try {
+    const fileStream = fs.createReadStream(assetFilePath);
+    const uploadedFile = await openAIClient.vectorStores.files.uploadAndPoll(
+      vectorStore.id,
+      fileStream,
+    );
+    console.log(`File uploaded to vector store (id: ${uploadedFile.id})`);
+    console.log("File processing completed");
+  } catch (error: any) {
+    console.log(`Warning: Could not upload file at ${assetFilePath}`);
+    console.log(`Error: ${error.message}`);
+    console.log("Creating vector store without file for demonstration...");
```

Comment:
why are we continuing the execution of the sample if this fails? Also, if we don't need to continue, we could ditch this error handling all together.